crates/bevy_winit/src/system.rs

```diff
@@ -101,6 +107,68 @@ pub fn create_windows<F: QueryFilter + 'static>(
     }
 }
 
+pub fn create_monitors(
+    event_loop: &ActiveEventLoop,
+    (mut commands, mut monitors): SystemParamItem<CreateMonitorParams>,
+) {
+    let primary_monitor = event_loop.primary_monitor();
+    let mut seen_monitors = vec![false; monitors.monitors.len()];
```

Comment:
How often is this called? If performance/allocations are not a blocker, a `BTreeSet<MonitorHandle>` might be nicer to use, you can then also build one from `monitors.monitors` and use `.difference()` to find the monitors no longer present.

Reply:
It's called very frequently, but in 98% of cases the user will only ever have a single monitor, so I doubt the performance differential is very much either way. I'm going to play around with one of the btree varients and see if that helps clean some of the code up. 👍

Reply:
Ah, okay, then maybe using smallvec would make a difference here? Not sure there is much need to optimize this too much without profiling but seeing a heap allocation for a list of 1-10 bools is sticking out :D

Reply:
The code is much cleaner with the btree, so I think it's a good idea!